compiler-rt/lib/hwasan/hwasan_report.cpp

```diff
@@ -754,7 +791,13 @@ void BaseReport::PrintAddressDescription() const {
            sa.thread_id());
     Printf("%s", d.Default());
     announce_by_id(sa.thread_id());
-    PrintStackAllocations(sa.get(), ptr_tag, untagged_addr);
+
+    if (common_flags()->enable_symbolizer_markup) {
+      PrintStackAllocationsMarkup(sa.get());
+    } else {
+      PrintStackAllocations(sa.get(), ptr_tag, untagged_addr);
+    }
+
```

Comment:
I think this particular function should remain untouched in hwasan. I believe this should just be able to call the various virtual symbolizer functions like `SymbolizePC` but it shouldn't care if markup is used or not.

Reply:
I've been thinking about this kind of functions and the separation of Markup vs no-Markup in reporting code. The main point of the separation is to not have markup context emitting code in not markup scenarios, otherwise code is almost identical. According to Roland, on the Fuchsia side, the code to emit context is called everytime an element is emitted. I could do something similar and call the functions to emit context from the `Render*` functions which will avoid changes in reporting code, like this one. My main problem with implementing that, is that the `Render*` functions now have to call the Symbolizer to get the modules and potentially other kind of context which could be an undesired dependency between the Rendering and the Symbolizer.